ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/BenchmarkReporter.scala

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.api.testtool.infrastructure
+
+import java.util.concurrent.atomic.AtomicReference
+
+trait BenchmarkReporter {
+
+  protected def addReport(key: String, value: Double): Unit
+
+  def forKey(parentKey: String)(key: String, value: Double): Unit =
+    addReport(parentKey + "." + key, value)
+
+}
+
+object BenchmarkReporter {
+  val toFile = new FileOutputBenchmarkReporter("benchmark.dat")
+}
+
+class FileOutputBenchmarkReporter(filename: String) extends BenchmarkReporter {
```

Comment:
Obviously this won't compile (you'll need to change the code below and the calling code), but I think this makes more sense:

```suggestion
class FileOutputBenchmarkReporter(path: Path) extends BenchmarkReporter {
```

Use `Paths.get` to construct a `Path`.